identity-apps-core/apps/authentication-portal/src/main/webapp/includes/branding-preferences.jsp

```diff
@@ -466,6 +486,66 @@
+
+                // Get the resolution strategy.
+                if (brandingPreference.has(THEME_KEY)) {
+                    if (brandingPreference.getJSONObject(THEME_KEY).has(RESOLUTION_STRATEGY_KEY)) {
+                        resolutionStrategy = brandingPreference.getJSONObject(THEME_KEY).getString(RESOLUTION_STRATEGY_KEY);
+                    }
+                }
+
+                // If the branding theme is set using the application.
+                if ("APP_PREFERENCE".equalsIgnoreCase(resolutionStrategy)) {
+
+                    // Check theme from the query param first.
+                    String resolvedUiTheme = request.getParameter("ui_theme");
+
+                    // If not in query param, check other parameteres.
+                    if (StringUtils.isBlank(resolvedUiTheme)) {
+                        String paramKey = null;
+
+                        if (StringUtils.isNotBlank(request.getParameter("callback"))) {
+                            paramKey = request.getParameter("callback");
+                        } else if (StringUtils.isNotBlank(request.getParameter("multiOptionURI"))) {
+                            paramKey = request.getParameter("multiOptionURI");
+                        } else if (StringUtils.isNotBlank(request.getParameter("urlQuery"))) {
+                            paramKey = request.getParameter("urlQuery");
+                        }
+
+                        if (StringUtils.isNotBlank(paramKey) && paramKey.contains("ui_theme=")) {
+                            try {
+                                String[] parts = paramKey.split("[?&]");
+                                for (String part : parts) {
+                                    if (part.startsWith("ui_theme=")) {
+                                        resolvedUiTheme = part.split("=")[1];
+                                        break;
+                                    }
+                                }
+                            } catch (Exception e) {
+                                // Silent fallback
+                            }
+                        }
+                    }
+
+                    // Fallback to cookie
+                    if (StringUtils.isBlank(resolvedUiTheme)) {
+                        resolvedUiTheme = currentCookieTheme;
+                    }
+
+                    uiThemeParam = resolvedUiTheme;
+
+                    // Write cookie only if resolved value is non-blank and different from cookie
+                    if (StringUtils.isNotBlank(resolvedUiTheme) && !resolvedUiTheme.equals(currentCookieTheme)) {
+                        
+                        String cookieName = "ui_theme";
+                        String cookieValue = resolvedUiTheme;
+                        int maxAge = 2592000;
+                    
+                        String headerValue = cookieName + "=" + cookieValue + "; Path=/; Max-Age=" + maxAge + "; Secure; HttpOnly; SameSite=None";
```

Comment:
This is setting the `ui_theme` cookie as a HTTP only cookie?
Hope that's expected. I believe now we are not accessing this cookie from JavaScript. But will we have to access it in JavaScript in the future?

Reply:
Since SameSite is set to None I added this for extra security so JS cant access this..as of now not accessing via JS..if in the future we need to read or modify it via JS, we can remove the HttpOnly flag at that time. wdyt?

Reply:
+1